### DISCUSSION TITLE
Impossibile recuperare lista Prodotti

### DISCUSSION THREAD

**Discussion by AndreaDev01** [fattureincloud/api#307] [Q&A] [answered]
Ciao, non riesco a recuperare la lista dei prodotti, anche inserendo negli scope il campo    `Scope::PRODUCTS_ALL,` ricevo sempre l'errore  GET https://api-v2.fattureincloud.it/c/id_company/products?page=1&per_page=5` resulted in a `401 Unauthorized` response: {"error":{"message":"Unauthorized."}}

**Comment by AndreaDev01** [comment]
Premetto che la chiamata listUserCompanies funziona correttamente

**Reply by valmoz** [reply]
Buongiorno @AndreaDev01 
Solitamente questo errore appare quando c'è un errore nell'impostazione dell'access token all'interno della richiesta, o se il token non è valido. Le chiedo quindi di controllare se il token è inserito nell'header come indicato da documentazione, oppure se sta usando una delle nostre SDK di controllare se è stata inizializzata correttamente.

Nel caso in cui non dovesse riuscire a risolvere, le chiedo gentilmente di scrivere al nostro supporto clienti. Purtroppo senza il company ID non è possibile aiutarla ulteriormente, e non è il caso di condividerlo in una community pubblica.

Grazie
Mauro

**Reply by valmoz** [reply]
[Support & Feature Request](https://developers.fattureincloud.it/docs/support)

**Reply by AndreaDev01** [reply]
Grazie per la risposta, ho provveduto a contattare il supporto clienti tramite chat, ancora non ricevo risposta

**Reply by valmoz** [reply]
Buongiorno.
Purtroppo visto l'elevato numero di richieste il supporto clienti ha tempi di risposta più lunghi.

La chiedo di pazientare, ed eventualmente di aggiungere il link a questa discussion in modo tale che i colleghi del supporto possano contattarmi a riguardo.

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @AndreaDev01.
Ho visto lo scambio con il nostro supporto clienti, è riuscito a risolvere il problema?

Grazie
Mauro

**Reply by AndreaDev01** [reply]
Buongiorno, si grazie

**Comment by AndreaDev01** [comment]
Il problema è stato risolto, l'errore era nell'inizializzazione sbagliata dell'oggetto

**Comment by valmoz** [comment, marked as answer]
Riporto la risposta per completezza.

Il problema era dovuto al fatto che la PHP SDK (come le altre) contiene diversi oggetti API, ciascuno dedicato ad un set di risorse specifico. Ciascun oggetto API deve essere inizializzato indipendentemente dagli altri.

Nel caso in questione, a differenza di altre API la _ProductsAPI_ non era stata configurata correttamente, e per questo motivo non aveva a disposizione il token per effettuare le chiamate.

Un esempio di inizializzazione può essere osservato sulla [repository della SDK](https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/docs/Api/ProductsApi.md#listproducts).
